File: Semester3/MAP/Seminare/Sem10/Sem10/Task/MessageTask.cs
namespace Sem10;

public class MessageTask : Task
{
    private string message, from, to;
    private DateTime date;

    public MessageTask(string id, string description, string message, string from, string to, DateTime date) : base(id, description)
    {
        this.message = message;
        this.from = from;
        this.to = to;
        this.date = date;
    }
    
    public override void execute()
    {
        Console.WriteLine($"Message: {message} | Date: {date}");
    }

    public override string ToString()
    {
        return base.ToString() + $"| Message: {message} | From: {from} | To: {to} | Date: {date}";
    }
}

File: Semester3/MAP/laboratoare/lab2/TaskManager/src/Domain/MessageTask.java
package Domain;

import javax.swing.text.DateFormatter;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.TimerTask;

public class MessageTask extends Task{
    private String message,from,to;
    private LocalDateTime date;
    public static DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");

    public MessageTask(String taskId, String description, String message, String from, String to, LocalDateTime date) {
        super(taskId, description);
        this.message = message;
        this.from = from;
        this.to = to;
        this.date = date;
    }

    @Override
    public void execute() {
        System.out.println(date.format(dateFormatter)+": "+message);
    }

    @Override
    public String toString() {
        return "Task: " + getTaskId() + " | Description: " + getDescription() + " | From: " + from + " | To: " + to + " | Date: " + date.format(dateFormatter);
    }
}


